shell/platform/windows/win32_flutter_window_unittests.cc

```diff
@@ -2,16 +2,461 @@
+        // "SendInput" directly to the window.
+        const KEYBDINPUT kbdinput = pInputs->ki;
+        const UINT message =
+            (kbdinput.dwFlags & KEYEVENTF_KEYUP) ? WM_KEYUP : WM_KEYDOWN;
+        const LPARAM lparam = CreateKeyEventLparam(
+            kbdinput.wScan, kbdinput.dwFlags & KEYEVENTF_EXTENDEDKEY);
+        // Windows would normally fill in the virtual key code for us, so we
+        // simulate it for the test with the key we know is in the test. The
+        // KBDINPUT we're passed doesn't have it filled in (on purpose, so that
+        // Windows will fill it in).
+        pending_events.push_back(SimulatedEvent{message, virtual_key, lparam});
+        if ((kbdinput.dwFlags & KEYEVENTF_KEYUP) == 0) {
+          pending_events.push_back(
+              SimulatedEvent{WM_CHAR, virtual_key, lparam});
+        }
+        return 1;
+      });
+  win32window.SetView(&flutter_windows_view);
+
+  // Test an event not handled by the framework
+  test_response = false;
+  flutter_windows_view.SetEngine(std::move(GetTestEngine()));
+  EXPECT_CALL(
+      *flutter_windows_view.key_event_handler,
+      KeyboardHook(_, virtual_key, scan_code, WM_KEYDOWN, character, false))
+      .Times(2);
+  EXPECT_CALL(*flutter_windows_view.text_input_plugin,
+              KeyboardHook(_, _, _, _, _, _))
+      .Times(1);
+  EXPECT_CALL(*flutter_windows_view.key_event_handler, TextHook(_, _)).Times(1);
+  EXPECT_CALL(*flutter_windows_view.text_input_plugin, TextHook(_, _)).Times(1);
+  LPARAM lparam = CreateKeyEventLparam(scan_code);
+  EXPECT_EQ(win32window.InjectWindowMessage(WM_KEYDOWN, virtual_key, lparam),
+            0);
+  EXPECT_EQ(win32window.InjectWindowMessage(WM_CHAR, virtual_key, lparam), 0);
+  while (pending_events.size() > 0) {
+    SimulatedEvent event = pending_events.front();
+    win32window.InjectWindowMessage(event.message, event.wparam, event.lparam);
+    pending_events.pop_front();
+  }
+  pending_events.clear();
+
+  // Test an event handled by the framework
+  test_response = true;
+  EXPECT_CALL(
+      *flutter_windows_view.key_event_handler,
+      KeyboardHook(_, virtual_key, scan_code, WM_KEYDOWN, character, false))
+      .Times(1);
+  EXPECT_CALL(*flutter_windows_view.text_input_plugin,
+              KeyboardHook(_, _, _, _, _, _))
+      .Times(0);
+  EXPECT_CALL(*flutter_windows_view.key_event_handler, TextHook(_, _)).Times(0);
+  EXPECT_CALL(*flutter_windows_view.text_input_plugin, TextHook(_, _)).Times(0);
+  EXPECT_EQ(win32window.InjectWindowMessage(WM_KEYDOWN, virtual_key, lparam),
+            0);
+  EXPECT_EQ(win32window.InjectWindowMessage(WM_CHAR, virtual_key, lparam), 0);
+  while (pending_events.size() > 0) {
+    SimulatedEvent event = pending_events.front();
+    win32window.InjectWindowMessage(event.message, event.wparam, event.lparam);
+    pending_events.pop_front();
+  }
+}
+
+// Tests key event propagation of modifier key down events.
```

Comment:
Can you tell me, or point me to, why they should behave differently? Is it because `SendEvent` doesn't simulate `WM_CHAR` events for them?